integration_tests/test_suites/daemon-test-suite/auto_run_reexecution_tests/test_auto_run_reexecution.py

```diff
@@ -422,6 +425,226 @@ def test_consume_new_runs_for_automatic_reexecution(instance, workspace_context)
+    # db but not the run_coordinator queue
+    instance.run_coordinator.queue().clear()
+    assert len(instance.run_coordinator.queue()) == 0
+    assert len(instance.get_runs()) == 2  # original run and retry run are both in the db
+
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+
+    # because the retried run already exists, it is in the run group for the original run. Because
+    # it is in the run group of the original run, we will see that the run group contains a run with
+    # parent_run_id as the original run. Therefore the original run will not be retried
+    # This is a problem, since the retried run will be stuck in a NOT_STARTED state
+    assert len(instance.run_coordinator.queue()) == 0
+    assert instance.get_run_by_id(first_retry_run_id).status == DagsterRunStatus.NOT_STARTED
+
+
+def test_consume_new_runs_for_automatic_reexecution_retry_run_deleted(instance, workspace_context):
+    instance.wipe()
+    instance.run_coordinator.queue().clear()
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+
+    assert len(instance.run_coordinator.queue()) == 0
+
+    run = create_run(instance, status=DagsterRunStatus.STARTED, tags={MAX_RETRIES_TAG: "2"})
+    dagster_event = DagsterEvent(
+        event_type_value=DagsterEventType.PIPELINE_FAILURE.value,
+        job_name="foo",
+        message="",
+    )
+    event_record = EventLogEntry(
+        user_message="",
+        level=logging.ERROR,
+        job_name="foo",
+        run_id=run.run_id,
+        error_info=None,
+        timestamp=time.time(),
+        dagster_event=dagster_event,
+    )
+    instance.handle_new_event(event_record)
+    run = instance.get_run_by_id(run.run_id)
+    assert run
+    assert run.tags.get(WILL_RETRY_TAG) == "true"
+
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+    assert len(instance.run_coordinator.queue()) == 1
+    first_retry = instance.run_coordinator.queue()[0]
+
+    # delete the run and remove it from the queue
+    instance.delete_run(first_retry.run_id)
+    instance.run_coordinator.queue().clear()
+    assert len(instance.run_coordinator.queue()) == 0
+    assert len(instance.get_runs()) == 1  # just the original run
+
+    list(
+        consume_new_runs_for_automatic_reexecution(
+            workspace_context,
+            instance.get_run_records(filters=RunsFilter(statuses=[DagsterRunStatus.FAILURE])),
+        )
+    )
+
+    assert len(instance.run_coordinator.queue()) == 1
```

Comment:
this one on the other hand you are definitely not going to fix right? (which i think is fine, this is a pretty unlikely edge case where you have to delete the run at exactly the wrong time)

maybe worth a comment explaining that this is a known race condition where things might not behave as you might expect

Reply:
added a comment, but you're correct, i'm not changing this behavior.